src/main/kotlin/org/rucca/cheese/space/analytics/SpaceAnalyticsService.kt

```
@@ -0,0 +1,291 @@
+    private fun generateStudentStatistics(
+        userIds: List<IdType>,
+        name: String,
+    ): StudentRealNameInfoStatisticsDTO {
+        val totalStudents = userIds.size
+
+        // 获取有真实姓名信息的学生
+        val studentsWithRealName =
+            userIds.mapNotNull { userId ->
+                try {
+                    userRealNameService.getUserIdentity(userId)
+                } catch (e: Exception) {
```

detekt (1.23.8)

[warning] The caught exception is swallowed. The original exception could be lost.

(detekt.exceptions.SwallowedException)